packages/react-app/src/views/BrowseBadges.jsx

```diff
@@ -145,20 +145,62 @@ export default function BrowseBadges() {
     run()
   }, [address, contractRef, eventBadges, localProvider, mainnet, selectedChainId])
 
+  const unwrap = async arr => {
+    const resolved = []
+    for (const badge of arr) {
+      resolved.push(await badge)
+    }
+    return resolved
+  }
+
   const run = useCallback(async () => {
     if (address) {
       return setEventBadges([])
     }
     let badges = await getAllRewards(contractRef.address, providerRef)
-    // console.log({ badges })
     badges = badges.map(badge => {
       return {
         id: ethers.utils.hexStripZeros(badge.topics[3]),
         to: ethers.utils.hexZeroPad(ethers.utils.hexStripZeros(badge.topics[2]), 20),
         transactionHash: badge.transactionHash,
       }
     })
+    let dataArray = []
+    const result = badges.map(async badge => {
+      if (badge.id === '0x') {
+        badge.id = '0x0'
+      }
+      let temp = { ...badge }
+
+      let data = await contract.current.tokensData(badge.id)
+      dataArray.push(data)
+      for (let index = 0; index < data.length; index++) {
+        if (data[index].length > 20) {
+          temp.hash = data[index]
+        }
+        if (data[index].length <= 15) {
+          temp.tokenType = data[index]
+        }
+        if (index === 0) temp.payload = data[index]
+      }
+
+      if (temp.tokenType === 'Release Manager') {
```

Comment:
why do you need to do that for the release manager badge and not for the betatester?

Reply:
Release Manager was the only reward that was not properly picking the payload. Some of the rewards have a blank payload. I just wanted to ensure that the payload is set with the right text

Reply:
that's strange: here's the data for the release manager: 

string: payload v0.26.0
string: tokenType Release Manager
bytes: hash 0x122008103e9586ade3d23d54fb75bf35a42b6ea538853c846be4d8f6a8263883986f

and of the beta testerr:

string: payload v0.24.0
string: tokenType Beta Tester
bytes: hash 0x12204064da946d65a2c966a50dfdb9506255e7ff52d2f971c9abc07dd5ce3f0e17e7

there shouldn't by any difference in the way there are processed.

Reply:
I understand your question now, because I am using the payload as a determinant of my reward definitions, I have to introduce a differentiator for release manager since it gets grouped with every other reward that belong under that payload.

Reply:
Changes to grouping determinant done. Thanks @yann300